Modules/_io/fileio.c

```diff
@@ -393,6 +393,11 @@ _io_FileIO___init___impl(fileio *self, PyObject *nameobj, const char *mode,
 
             if (async_err)
                 goto error;
+
+            if (self->fd < 0) {
+                PyErr_SetFromErrnoWithFilenameObject(PyExc_OSError, nameobj);
+                goto error;
+            }
```

Comment:
``fd_is_own = 1;`` changes the code in the error label, it's unclear to me if your change is correct or not.

Reply:
Ok, I see. This change is non-trivial, I preferred to double check.